docs/tooling/cli/04-templates.md

```diff
@@ -0,0 +1,149 @@
+---
+id: templates
+title: CLI Templates
+description: Overview of the new CLI Declarative Templates
+---
+
+The behavior of the `backstage-cli new` command is configurable through your root `package.json`, and you can also create and add custom CLI templates to suit your needs.
+
+## Basic Configuration
+
+```json
+{
+  "name": "root",
+  "backstage": {
+    "cli": {
+      "new": {
+        "globals": {
+          "license": "MIT",
+          "namePrefix": "@my-org/"
+        }
+      }
+    }
+  }
+}
+```
+
+- `globals` - Configures input for all generated packages and plugins.
+  - `version` - Sets the value of the `version` field in `package.json` of all generated packages. Defaults to `Apache-2.0`.
+  - `license` - Sets the value of the `license` field in `package.json` of all generated packages. Defaults to `0.1.0`.
+  - `private` - Sets the value of the `private` field in `package.json` of all generated packages. Defaults to `true`.
+  - `publishRegistry` - Sets the value of the `publishConfig.registry` field in `package.json` of all generated packages.
+  - `namePrefix` - The prefix used to generate the full package name. Defaults to `@internal/`.
+  - `namePluginInfix` - The infix used to generate the full package name for plugin packages. Defaults to `plugin-`.
+- `templates` - Specifies custom templates.
+  - See [Installing custom templates](#installing-custom-templates) and [Creating your own CLI templates](#creating-your-own-cli-templates) for more information.
+
+The generated package name is based on the `namePrefix` and `namePluginInfix` globals, as well as the "base name" which is derived from the package role and user input. For plugin packages the final package name will be `<namePrefix><namePluginInfix><baseName>`, and for other packages it will be `<namePrefix><baseName>`.
+
+For example, if you want your plugin frontend packages to end up with the name `@acme/backstage-plugin-<pluginId>`, you should use the following configuration:
+
+```json
+{
+  "name": "root",
+  "backstage": {
+    "cli": {
+      "new": {
+        "globals": {
+          "namePrefix": "@acme/",
+          "namePluginInfix": "backstage-plugin-"
+        }
+      }
+    }
+  }
+}
+```
+
+## Installing custom templates
+
+Custom templates can be installed from local directories. To install a template you add it to the `backstage.cli.new.templates` configuration array in your root `package.json`:
+
+```json
+{
+  "name": "root",
+  "backstage": {
+    "cli": {
+      "new": {
+        "templates": ["./templates/custom-plugin"]
+      }
+    }
+  }
+}
+```
+
+Each entry in the `templates` array should be relative path that points to a directory containing a `portable-template.yaml` file. If the path starts with `./` it will be use as is, otherwise it will be resolved as a module within `node_modules`.
```

Comment:
```suggestion
Each entry in the `templates` array should be relative path that points to a directory containing a `portable-template.yaml` file. If the path starts with `./` it will be used as is, otherwise it will be resolved as a module within `node_modules`.
```